### ISSUE TITLE
🚀 Ability to apply links for circular progress tracker.(advanced back to top button)

### ISSUE THREAD

**Issue by 79ho3ein** [elementor/elementor#18327]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

It would be great if you could add a link to the circular progress tracker. By adding this feature, we can create a very attractive back to top button. https://prnt.sc/4DNhDDrmTinT
It will also be great to be able to insert a [custom icon in the center of the circular progress tracker](https://prnt.sc/1d5SOUo0NWAw).
Another thing is that the circular progress track has empty space around it by default and is not controllable. https://prnt.sc/Ru_hj9qHCcTm

Finally, by adding these two features, we can create a very advanced back to top button in elementor.

### Describe the solution you'd like

advanced back to top button

**Comment by rami-elementor** [comment]
Hi @79ho3ein 

Like the one you mentioned in #16261 ?

**Comment by 79ho3ein** [comment]
@rami-elementor 

No, this is actually very different from the progress bar widget.
Using this feature, we can turn the circular progress tracker into a back to top button. But here an interesting thing happens! When the user scrolls down the page, the circular progress tracker displays the user's scrolling rate. When the user clicks on the circular progress tracker, it will be moved to the top of the page and that is really great. I may not have conveyed my point well, but I'm trying to find a sample to show you in a video.

**Comment by 79ho3ein** [comment]
hi again @rami-elementor , Please watch the video below ,This is exactly what I mentioned in this issue. 


https://user-images.githubusercontent.com/77163248/162620841-48772624-b7ec-44db-84a5-1c238c492fe8.mp4

This feature has two advantages :

1- Is the Advanced back to top button
2- Like the progress tracker widget, it displays the user's scrolling rate.


I think in general, elementor desperately needs a new, professional back to top widget. For various reasons, which I will mention below ...
1- Using this widget will be much easier and more professional for users.
2-It will be much easier to develop this feature in the future.
3- If you decide to add the feature you saw in the video above, developing it in this new widget will be much easier than the progress tracker widget.
4- In this widget, we can add great features (I will explain in the end of item 5)
5- In this [issue you](https://github.com/elementor/elementor/issues/18332) said that **we can use id to add the back to top attribute.** But many users are not aware of this trick. Also, this method is very, very simple and we can not have features like the following in it.
A : The back to top button is hidden first and appears after 20 pixels of scrolling.
B : After displaying this button, the button will be hidden when the user does not scroll for x seconds. (Appears after scrolling again)
C : Entrance Animation & Exit Animation. (It has animation when the button appears. For example, in the form of Woodmart, the back to top button has this animation.)
D : Ability to set the speed (smooth) in this button (Determine how fast the user moves up when they click this button.)

I think adding this widget would be great. Users are also very keen on this feature. I have been working with elementor for more than 4 years and I interact with a large community of its users and I am aware of their needs. 

Anyway, thank you!

**Comment by 79ho3ein** [comment]
This great feature is also implemented in the element pack plugin : https://www.elementpack.pro/demo/element/reading-progress/back-to-top-progress/
